portal/src/main/webapp/js/src/study-view/component/PieChart.js

```diff
@@ -960,21 +967,22 @@ var PieChart = function(){
 
     //Called when the number of label biggen than 6, used by addPieLabels()
     function labelFunction() {
-        var _tableDiv = '';
+        var _tableDiv = [];
+        var _tableDivStr = '';
         var _labelLT = labelWLT; //label length threshold
         var _labelHeaderLT = labelHeaderWLT; //label header length threshold
 
-        _tableDiv += '<table';
+        _tableDiv.push('<table');
```

Comment:
I changed from appendTo actually. jQuery appendTo is really slow sometimes especially when you try to append several times. Instead, building up an array could be one of better solution for this.